feat: Product card not found: ensure the content fits the screen

g123k opened the pull request:
### What
- On small devices, the Product not found card may overflow the screen

Changes in this PR:
- Include a `FittedBox`
- Move the 3 `Texts` into a single `RichText`

### Screenshot
(On an iPod Touch)

![Simulator Screen Shot - iPod touch (7th generation) - 2022-09-06 at 15 57 30](https://user-images.githubusercontent.com/246838/188656299-8ca1d444-3eb2-4a4a-978b-041434dabfcf.png)


Will fix #2946

g123k committed:
Product card not found: ensure the content fits the screen

monsieurtanuki reviewed (commented):
You're cheating: this is not the same text! ;)
cf. `add_product_take_photos`

teolemon reviewed (approved):
honestly, it works

monsieurtanuki commented:
@teolemon Indeed, if you don't display the same text and take a shorter one, yes it works. As the text is now repeated it would even have been faster to remove the second one.

g123k committed:
Product card not found: ooops, wrong text use for the second line

g123k commented:
Oops you're right 🙈 it's now fixed.
Please let me a few minutes because the invalid barcode card also has problems.

Fixed screenshot:
![Simulator Screen Shot - iPod touch (7th generation) - 2022-09-07 at 07 43 34](https://user-images.githubusercontent.com/246838/188798022-f86400d6-b43e-4e98-9d68-665fd14a3ea7.png)

g123k committed:
Common layout for welcome / product not found / error cards

g123k commented:
I prefer to create a new PR, as this one is limited to one card
